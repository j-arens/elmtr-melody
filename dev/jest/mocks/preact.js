module.exports = {
    unmountComponent: jest.fn(),
};
