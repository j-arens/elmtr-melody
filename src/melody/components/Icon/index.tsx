import { WithOptionalClassName } from '@components/type';
import { h } from 'preact';
import BufferMarkup from './svg/buffer';
import DownloadMarkup from './svg/download';
import ErrorMarkup from './svg/error';
import MinusMarkup from './svg/minus';
import NextMarkup from './svg/next';
import PauseMarkup from './svg/pause';
import PlayMarkup from './svg/play';
import PlusMarkup from './svg/plus';
import PrevMarkup from './svg/prev';
import RepeatMarkup from './svg/repeat';
import SettingsMarkup from './svg/settings';
import ShuffleMarkup from './svg/shuffle';

interface Props extends WithOptionalClassName {
    name: IconType;
}

type IconType =
    | 'shuffle'
    | 'prev'
    | 'play'
    | 'pause'
    | 'next'
    | 'repeat'
    | 'buffer'
    | 'error'
    | 'download'
    | 'plus'
    | 'minus'
    | 'settings';

export default ({ name, className = '' }: Props) => {
    const getMarkup = (icon: IconType): JSX.Element => {
        switch (icon) {
            case 'shuffle': {
                return ShuffleMarkup;
            }
            case 'prev': {
                return PrevMarkup;
            }
            case 'play': {
                return PlayMarkup;
            }
            case 'pause': {
                return PauseMarkup;
            }
            case 'next': {
                return NextMarkup;
            }
            case 'repeat': {
                return RepeatMarkup;
            }
            case 'buffer': {
                return BufferMarkup;
            }
            case 'error': {
                return ErrorMarkup;
            }
            case 'download': {
                return DownloadMarkup;
            }
            case 'plus': {
                return PlusMarkup;
            }
            case 'minus': {
                return MinusMarkup;
            }
            case 'settings': {
                return SettingsMarkup;
            }
        }
    };

    return (
        <svg
            class={className}
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
        >
            {getMarkup(name)}
        </svg>
    );
};
