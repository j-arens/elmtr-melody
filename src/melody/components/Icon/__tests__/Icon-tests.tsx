import { render } from 'preact-render-to-string';
import { h } from 'preact';
import Icon from '../';

describe('Icon', () => {
    it('renders', () => {
        const component = render(
            <Icon className="jet-black" name="" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders shuffle icon', () => {
        const component = render(
            <Icon className="jet-black" name="shuffle" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders prev icon', () => {
        const component = render(
            <Icon className="jet-black" name="prev" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders play icon', () => {
        const component = render(
            <Icon className="jet-black" name="play" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders pause icon', () => {
        const component = render(
            <Icon className="jet-black" name="pause" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders next icon', () => {
        const component = render(
            <Icon className="jet-black" name="next" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders repeat icon', () => {
        const component = render(
            <Icon className="jet-black" name="repeat" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders buffer icon', () => {
        const component = render(
            <Icon className="jet-black" name="buffer" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders error icon', () => {
        const component = render(
            <Icon className="jet-black" name="error" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders download icon', () => {
        const component = render(
            <Icon className="jet-black" name="download" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders plus icon', () => {
        const component = render(
            <Icon className="jet-black" name="plus" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders minus icon', () => {
        const component = render(
            <Icon className="jet-black" name="minus" />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders settings icon', () => {
        const component = render(
            <Icon className="jet-black" name="settings" />
        );
        expect(component).toMatchSnapshot();
    });
});
