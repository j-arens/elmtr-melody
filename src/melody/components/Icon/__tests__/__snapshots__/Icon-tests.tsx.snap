// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders 1`] = `<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"></svg>`;

exports[`Icon renders buffer icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <g>
    <circle class="melody-playbackCtrl__ring" cx="12" cy="12" r="12"></circle>
    <g class="melody-playbackCtrl__buffer-dots">
      <circle cx="4" cy="13" r="2"></circle>
      <circle cx="12" cy="13" r="2"></circle>
      <circle cx="20" cy="13" r="2"></circle>
    </g>
  </g>
</svg>
`;

exports[`Icon renders download icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"></path>
</svg>
`;

exports[`Icon renders error icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="m110 150 20 0 0 20-20 0zm0-80 20 0 0 60-20 0zm9.9-50C64.7 20 20 64.8 20 120 20 175.2 64.7 220 119.9 220 175.2 220 220 175.2 220 120 220 64.8 175.2 20 119.9 20Zm0.1 180c-44.2 0-80-35.8-80-80 0-44.2 35.8-80 80-80 44.2 0 80 35.8 80 80 0 44.2-35.8 80-80 80z"></path>
</svg>
`;

exports[`Icon renders minus icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M19 13H5v-2h14v2z"></path>
</svg>
`;

exports[`Icon renders next icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M4 18l8.5-6L4 6v12zm9-12v12l8.5-6L13 6z"></path>
</svg>
`;

exports[`Icon renders pause icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <g>
    <circle class="melody-playbackCtrl__ring" cx="12" cy="12" r="12"></circle>
    <path d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"></path>
  </g>
</svg>
`;

exports[`Icon renders play icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <g>
    <circle class="melody-playbackCtrl__ring" cx="12" cy="12" r="12"></circle>
    <path d="M8 5v14l11-7z"></path>
  </g>
</svg>
`;

exports[`Icon renders plus icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"></path>
</svg>
`;

exports[`Icon renders prev icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M11 18V6l-8.5 6 8.5 6zm.5-6l8.5 6V6l-8.5 6z"></path>
</svg>
`;

exports[`Icon renders repeat icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"></path>
</svg>
`;

exports[`Icon renders settings icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"></path>
</svg>
`;

exports[`Icon renders shuffle icon 1`] = `
<svg class="jet-black" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"></path>
</svg>
`;
